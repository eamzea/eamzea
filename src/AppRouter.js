import React, { useEffect, useReducer } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import PortfolioApp from './apps/Portfolio/PortfolioApp';
import DigitalKeyApp from './apps/DigitalKey/DigitalKeyApp';
import PromasaSJTApp from './apps/PromasaSJT/PromasaSJTApp';
import HeroesApp from './apps/Heroes/HeroesApp';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'animate.css';

const AppRouter = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/digital-key" component={DigitalKeyApp} />
        <Route path="/promasaSJT" component={PromasaSJTApp} />
        <Route path="/heroes" component={HeroesApp} />
        <Route path="/" component={PortfolioApp} />
      </Switch>
    </Router>
  );
};

export default AppRouter;
