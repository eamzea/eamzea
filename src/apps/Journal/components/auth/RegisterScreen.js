import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { setError, removeError } from '../../actions/ui';
import { startRegisterEmailPasswordName } from '../../actions/auth';

const RegisterScreen = () => {
  const dispatch = useDispatch();

  const { msgError } = useSelector(state => state.ui);

  const [formValues, handleInputChange] = useForm({
    name: 'Edgar',
    email: 'eamzea@gmail.com',
    password: 'cesarZEA210394',
    password2: 'cesarZEA210394',
  });

  const { name, email, password, password2 } = formValues;

  const handleRegister = e => {
    e.preventDefault();

    if (isFormValid()) {
      dispatch(startRegisterEmailPasswordName(email, password, name));
    }
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setError('Name is Required'));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setError('Email is not valid'));
      return false;
    } else if (password.length <= 5) {
      dispatch(setError('Password should be at least 6 characters'));
      return false;
    } else if (password !== password2) {
      dispatch(setError(`Passwords don't match`));
      return false;
    }

    dispatch(removeError());

    return true;
  };

  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form
        onSubmit={handleRegister}
        className="animate__animated animate__fadeIn animate__faster"
      >
        {msgError && <div className="auth__alert-error">{msgError}</div>}

        <input
          className="auth__input"
          type="text"
          placeholder="Name"
          name="name"
          autoComplete="off"
          value={name}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="text"
          placeholder="Email"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Confirm password"
          name="password2"
          value={password2}
          onChange={handleInputChange}
        />
        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>

        <Link to="/journal/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </>
  );
};

export default RegisterScreen;
