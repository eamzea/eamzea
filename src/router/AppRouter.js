import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import PortfolioApp from '../apps/Portfolio/PortfolioApp';
import DigitalKeyApp from '../apps/DigitalKey/DigitalKeyApp';
import PromasaSJTApp from '../apps/PromasaSJT/PromasaSJTApp';
import HeroesApp from '../apps/Heroes/HeroesApp';
import JournalApp from '../apps/Journal/JournalApp';
import 'animate.css';

const AppRouter = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/digital-key" component={DigitalKeyApp} />
        <Route path="/promasaSJT" component={PromasaSJTApp} />
        <Route path="/heroes" component={HeroesApp} />
        <Route path="/journal" component={JournalApp} />
        <Route path="/" component={PortfolioApp} />
      </Switch>
    </Router>
  );
};

export default AppRouter;
