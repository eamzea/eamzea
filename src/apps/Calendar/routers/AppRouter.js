import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { startChecking } from '../actions/auth';
import LoginScreen from '../components/auth/LoginScreen';
import CalendarScreen from '../components/calendar/CalendarScreen';
import LoadingScreen from '../components/ui/LoadingScreen';
import { PrivateRoutes } from './PrivateRoute';
import { PublicRoutes } from './PublicRoute';

const AppRouter = () => {
  const dispatch = useDispatch();

  const { checking, uid } = useSelector(state => state.auth);

  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);

  if (checking) {
    return <LoadingScreen />;
  }

  return (
    <Router>
      <div style={{ width: '100vw' }}>
        <Switch>
          <PublicRoutes
            path="/calendar/login"
            component={LoginScreen}
            isAuthenticated={!!uid}
          />
          <PrivateRoutes
            exact
            path="/calendar"
            component={CalendarScreen}
            isAuthenticated={!!uid}
          />
          <Redirect to="/calendar/login" />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
