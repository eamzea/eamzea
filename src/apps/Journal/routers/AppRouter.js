import React, { useEffect, useState } from 'react';
import { Switch, BrowserRouter as Router, Redirect } from 'react-router-dom';
import AuthRouter from './AuthRouter';
import JournalScreen from '../components/journal/JournalScreen';
import { firebase } from '../firebase/firebaseConfig';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';
import LoadingScreen from '../components/ui/LoadingScreen';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';
import { startLoadingNotes } from '../actions/notes';

const AppRouter = () => {
  const dispatch = useDispatch();

  const [checking, setChecking] = useState(true);

  const [isLoggedIn, setIsLoggedIn] = useState(false);

  useEffect(() => {
    firebase.auth().onAuthStateChanged(async user => {
      if (user?.uid) {
        dispatch(login(user.uid, user.displayName));
        setIsLoggedIn(true);

        dispatch(startLoadingNotes(user.uid));
      } else {
        setIsLoggedIn(false);
      }
      setChecking(false);
    });
  }, [dispatch, setChecking, setIsLoggedIn]);

  if (checking) {
    return <LoadingScreen />;
  }

  return (
    <Router>
      <div style={{ width: '100vw' }}>
        <Switch>
          <PublicRoutes
            isAuthenticated={isLoggedIn}
            path="/journal/auth"
            component={AuthRouter}
          />
          <PrivateRoutes
            exact
            isAuthenticated={isLoggedIn}
            path="/journal"
            component={JournalScreen}
          />
          <Redirect to="/journal/auth/login" />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
