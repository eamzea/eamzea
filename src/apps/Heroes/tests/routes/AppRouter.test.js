import "@testing-library/jest-dom";
import React from "react";
import { shallow, mount } from "enzyme";
import AppRouter from "../../routers/AppRouter";
import { AuthContext } from "../../auth/AuthContext";

describe("Pruebas en <AppRouter/>", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  test("Debe de mostrar login si no está autenticado", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el componente <Marvel/> si está autenticado", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
        name: "Edgar",
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find("nav").exists()).toBe(true);
  });
});
