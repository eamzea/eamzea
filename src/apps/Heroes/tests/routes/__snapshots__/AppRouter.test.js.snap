// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Debe de mostrar login si no est√° autenticado 1`] = `
<div>
  <div
    className="mt-5 p-5 vh-100 container-fluid"
  >
    <div
      className="justify-content-center h-100 row"
    >
      <div
        className="d-flex flex-column justify-content-center align-items-center col-6"
      >
        <h1>
          Login Screen
        </h1>
        <hr />
        <button
          className="btn btn-primary btn-lg"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </div>
    </div>
  </div>
</div>
`;
